### PR TITLE
[agent] Add Spark Agent

### PR DESCRIPTION
**Pull request by skcoirz**
- added support for spark through pyspark library.
- added jupyter notebook as example.

### CODE REVIEW

File: langchain/agents/agent_toolkits/spark/prompt.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# flake8: noqa
+
+PREFIX = """
+You are working with a spark dataframe in Python. The name of the dataframe is `spdf`.
```
Comment (gengliangwang) [inline langchain/agents/agent_toolkits/spark/prompt.py:4]: Nit: shall we make the dataframe name as `df`? Probably you are trying to make a difference from the panda dataframe, but many Spark users are using `df`.

**Comment by skcoirz** [comment]
No prob! Will change.

**Review by gengliangwang — APPROVED** [review]
LGTM since the code is similar to the pandas dataframe agent

File: docs/modules/agents/toolkits/examples/spark.ipynb [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "from langchain.llms import OpenAI\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "spark = SparkSession.builder.getOrCreate()\n",
+    "csv_file_path = \"titanic.csv\"\n",
+    "spdf = spark.read.csv(csv_file_path, header=True, inferSchema=True)\n",
```
Comment (allisonwang-db) [inline docs/modules/agents/toolkits/examples/spark.ipynb:70]: nit: we can use `df` here

File: docs/modules/agents/toolkits/examples/spark.ipynb [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+      "\n",
+      "\n",
+      "\u001b[1m> Entering new AgentExecutor chain...\u001b[0m\n",
+      "\u001b[32;1m\u001b[1;3mThought: I need to get the average age first\n",
+      "Action: python_repl_ast\n",
+      "Action Input: spdf.agg({\"Age\": \"mean\"}).collect()\u001b[0m\n",
+      "Observation: \u001b[36;1m\u001b[1;3m[Row(avg(Age)=29.69911764705882)]\u001b[0m\n",
+      "Thought:\u001b[32;1m\u001b[1;3m I need to get the square root of the average age\n",
+      "Action: python_repl_ast\n",
+      "Action Input: math.sqrt(29.69911764705882)\u001b[0m\n",
```
Comment (allisonwang-db) [inline docs/modules/agents/toolkits/examples/spark.ipynb:175]: Here the agent first tries to get the average age in using spark, then uses `math.sqrt` to compute it. Another approach is to directly use the pyspark SQL function `sqrt`. Is there a way to make the agent use the SQL function?

Comment (skcoirz) [reply docs/modules/agents/toolkits/examples/spark.ipynb:175]: Yeah I have a plan to add more features here for spark agent. This is the initial commit. I need to experiment more before adding next feature. Thank you for the suggestion!

**Review by allisonwang-db — APPROVED** [review]

**Review by hwchase17 — APPROVED** [review]
looks great! thanks

**Pull request merged**